ClemBot.Bot/bot/cogs/trivia_cog.py

```diff
@@ -0,0 +1,566 @@
+            x += 1
+
+        difficulty_generator = helper_fixer(DIFFICULTY)
+
+        for i in difficulty_generator:
+            embed_list.append(i)
+
+        new_insertion = len(embed_list)
+
+        while sizeofcategory < new_insertion:
+            difficulty_embed = discord.Embed(title="Difficulty List:", color=Colors.ClemsonOrange)
+            difficulty_embed.add_field(name="Index:", value=embed_list[sizeofcategory])
+
+            final_page.append(difficulty_embed)
+            sizeofcategory += 1
+
+        question_generator = helper_fixer(QUESTIONTYPE)
+
+        for i in question_generator:
+            embed_list.append(i)
+
+        last_size = len(embed_list)
+        while new_insertion < last_size:
+            type_embed = discord.Embed(title="Question Type:", color=Colors.ClemsonOrange)
+            type_embed.add_field(name="Index:", value=embed_list[new_insertion])
+
+            final_page.append(type_embed)
+            new_insertion += 1
+
+        await self.bot.messenger.publish(Events.on_set_pageable_embed,
+                                         pages=final_page,
+                                         author=ctx.author,
+                                         channel=ctx.channel,
+                                         timeout=60, )
+
+        return
+
+    async def url_builder(self, function_list, input_length):
+
+        max_index = input_length - 1
+        url = URL_BUILDER + str(function_list[0])
+        x = 1
+
+
+        while x <= max_index:
+            if function_list[x]:
+                match x:
+                    case 1:
+                        url += ("&category=" + str(function_list[1]))
+                    case 2:
+                        url += ("&difficulty=" + function_list[2])
+                    case 3:
+                        url += ("&type=" + function_list[max_index])
+            x += 1
+
+        return url
+
+    async def matching_function(self, case, *input: str):
+
+        match case:  #Revolves around beautiful O(1) based indexing
+            case 0:
+                if input[0].isnumeric():
+                    question_number = int(input[0])
+
+                    if 0 < question_number <= 50:
+                        return question_number
+                else:
+                    raise UserInputError(
+                        "Question Number has to be a number within the range of 1 to 50")
+            case 1:
+                if input[1].isnumeric():
+                    trivia_number = int(input[1])
+                    if 0 < trivia_number <= 24:
+                        return trivia_number + 8
+                    elif trivia_number == 0:
+                        return None
+                    else:
+                        raise UserInputError(
+                            "Category Number out of bounds(Number has to be 1-24) or enter the category you want! Type ?trivia help to see the category list")
```

Comment:
We need to grab the actual command prefix for the context guild and interpolate it into this string